photutils/psf/daogroup.py

```diff
@@ -0,0 +1,129 @@
+    ----------
+    starlist : `~astropy.table.Table` or array-like
+        List of stars positions.
+        If `~astropy.table.Table`, columns should be named 'x_0' and 'y_0'.
+        TODO: If array-like, it should be either (x_0, y_0).
+    crit_separation : float or int
+        Distance, in units of pixels, such that any two stars separated by
+        less than this distance will be placed in the same group.
+
+    Returns
+    -------
+    group_starlist : list of `~astropy.table.Table`
+        Each `~astropy.table.Table` in the list corresponds to a group of
+        mutually overlapping starts.
+
+    Notes
+    -----
+    Assuming the psf fwhm to be known, 'crit_separation' may be set to
+    k*fwhm, for some positive real k.
+
+    See
+    ---
+    `~daofind`
+    """
+
+    if not isinstance(crit_separation, (float, int)):
+        raise ValueError('crit_separation is expected to be either float or' +
+                         ' int. Received {}.'.format(type(crit_separation)))
+    if crit_separation < 0.0:
+        raise ValueError('crit_separation is expected to be a positive' +
+                         'real number.')
+
+    ## write a method that varifies whether the starlist given by the user
+    ## is valid
+
+    group_starlist = []
+    cstarlist = starlist.copy()
+
+    if 'id' not in cstarlist.colnames:
+        cstarlist.add_column(Column(name='id',
+                                    data=np.arange(len(cstarlist))))
+    
+    while len(cstarlist) is not 0:
+        init_group = _find_group(cstarlist[0], cstarlist, crit_separation)
+        assigned_stars_ids = np.intersect1d(cstarlist['id'], init_group['id'],
+                                            assume_unique=True)
+        cstarlist = _remove_stars(cstarlist, assigned_stars_ids)
+        n = 1
+        N = len(init_group)
+        while(n < N):    
+            tmp_group = _find_group(init_group[n], cstarlist, crit_separation)
+            if len(tmp_group) > 0:
+                assigned_stars_ids = np.intersect1d(cstarlist['id'],
+                                                    tmp_group['id'],
+                                                    assume_unique=True)
+                cstarlist = _remove_stars(cstarlist, assigned_stars_ids)
+                init_group = vstack([init_group, tmp_group])
+                N = len(init_group)
+            n = n + 1
+        group_starlist.append(init_group)
+    return group_starlist
+
+def _find_group(star, starlist, crit_separation):
+    """
+    Find those stars in `starlist` which are at a distance of
+    `crit_separation` from `star`.
+
+    Parameters
+    ----------
+    star : `~astropy.table.row.Row`
+        Star which will be either the head of a cluster or an isolated one.
+    
+    starlist : `~astropy.table.table.Table`
+
+    Returns
+    -------
+    `~astropy.table.table.Table` containing those stars which are at a distance
+    of `crit_separation` from `star`.
+    """
+    
```

Comment:
Just thought I'd ask whether the
scipy.spatial.KDTree.query_pairs
method could be used here to identify all the pairs with dist < crit_separation